synthesizer/program/src/logic/instruction/operation/hash.rs

```diff
@@ -256,14 +462,9 @@ impl<N: Network, const VARIANT: u8> HashInstruction<N, VARIANT> {
             (_, PlaintextType::Struct(..)) => bail!("Cannot hash into a struct"),
             (_, PlaintextType::Array(..)) => bail!("Cannot hash into an array (yet)"),
         };
-        // Cast the output to the destination type.
-        let output = match self.destination_type {
```

Comment:
Generally, it would nice to have fewer deltas on the variants that are not being changed.